### PR TITLE
Avoid adding product more than once and add total

### PR DESCRIPTION
**Pull request by valevassallo**
In this PR:
- the addition of a product to the cart more than once from the restaurant view is avoided
- the total price is added to the cart

### CODE REVIEW

File: src/reducer.js [hunk 1]
```diff
@@ -50,6 +50,13 @@ function reducer(state = initialState, action = {}) {
             }
           }
         };
+      } else {
```
Comment (CesarCachay) [inline src/reducer.js]: This line allow us to validate if we don't have an order. But if we have it show the order made right?

File: src/selectors.js [hunk 2]
```diff
@@ -20,10 +20,19 @@ function useMenutItems() {
   return useSelector(state => Object.values(state.menuItems), shallowEqual);
 }
 
+function useGetTotal() {
+  return useSelector(state =>
+    Object.values(state.cart).reduce((acc, item) => {
+      return acc + item.quantity * item.price;
+    }, 0)
```
Comment (CesarCachay) [inline src/selectors.js]: Could you explain why did you use a reducer and finish with 0?
What about changing that 0 to an empty object `{}`?

Comment (valevassallo) [reply src/selectors.js]: I used the reducer to add all the prices at the end, and I used the zero as the initial value of the accumulator, you want to start counting from 0, not from an empty object

File: src/reducer.js (hunk 1 above)
Comment (valevassallo) [reply src/reducer.js]: No, it's actually the other way round.

**Pull request merged**